components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java

```diff
@@ -150,26 +150,31 @@ private String getLogoutReturnUrl(ServiceProvider application) {
     private AssociatedRolesConfig buildAssociatedRoles(ServiceProvider application) {
 
         AssociatedRolesConfig associatedRolesConfig = new AssociatedRolesConfig();
-        if (application.getAssociatedRolesConfig() == null) {
+        org.wso2.carbon.identity.application.common.model.AssociatedRolesConfig associatedRolesConfiguration =
+                application.getAssociatedRolesConfig();
+        if (associatedRolesConfiguration == null) {
             return null;
         }
 
-        String allowedAudience = application.getAssociatedRolesConfig().getAllowedAudience();
+        String allowedAudience = associatedRolesConfiguration.getAllowedAudience();
         AssociatedRolesConfig.AllowedAudienceEnum allowedAudienceEnum = null;
+
         switch (allowedAudience) {
-            case "application":
+            case RoleConstants.APPLICATION:
                 allowedAudienceEnum = AssociatedRolesConfig.AllowedAudienceEnum.APPLICATION;
                 break;
-            case "organization":
+            case RoleConstants.ORGANIZATION:
                 allowedAudienceEnum = AssociatedRolesConfig.AllowedAudienceEnum.ORGANIZATION;
                 break;
             default:
                 break;
         }
         associatedRolesConfig.setAllowedAudience(allowedAudienceEnum);
-        RoleV2[] roles = application.getAssociatedRolesConfig().getRoles();
-        Arrays.asList(roles).forEach(role -> associatedRolesConfig.addRolesItem(
-                new Role().id(role.getId()).name(role.getName())));
+        if (RoleConstants.APPLICATION.equals(allowedAudience)) {
+            Arrays.stream(associatedRolesConfiguration.getRoles())
+                    .map(role -> new Role().id(role.getId()).name(role.getName()))
+                    .forEach(associatedRolesConfig::addRolesItem);
+        }
```

Comment:
why we are not setting organization audience roles here?

Reply:
The intention of this feature is to remove the association of the organization roles to a particular application. Hence we need to associate only the application roles.